### ISSUE TITLE
Error "No Newsletter matches the given query." depending on site setting of newsletter

### ISSUE THREAD

**Issue by TommyK82** [jazzband/django-newsletter#354]
HI,

I found your app, which is really  amazing.

Anyway  I am reading the docs for 2 days and  cant  find an answer.

I installed  everything like in the docs   (I am using  django 3.1.3 ; django-newsletter 0.9.1; python 3.8.5)

Newsletter sign up works when I set "example.com"  site for the newsletter "test3":
[06/Dec/2020 19:32:19] "GET /admin/logout/ HTTP/1.1" 200 1519
[06/Dec/2020 19:32:22] "GET / HTTP/1.1" 200 11928
[06/Dec/2020 19:32:28] "POST /newsletter/test3/subscribe/ HTTP/1.1" 302 0
[06/Dec/2020 19:32:28] "GET /newsletter/test3/subscribe/email-sent/ HTTP/1.1" 200 469

But If I set the localhost "http://127.0.0.1:8000/" as  site for the newsletter "test3"  then nothing works
Not Found: /newsletter/test3/subscribe/
[06/Dec/2020 19:38:45] "POST /newsletter/test3/subscribe/ HTTP/1.1" 404 1787

Error-page:
Page not found (404)
Request Method: 	POST
Request URL: 	http://127.0.0.1:8000/newsletter/test3/subscribe/
Raised by: 	newsletter.views.SubscribeRequestView

No Newsletter matches the given query.

Any clue?  Thanks  a lot

P.S.:  another "issue" I have  is  copying the ‘stub’-templates to  the  project_dir  does not change  a thing, there are still the  the templates form the package linked

**Comment by TommyK82** [comment]
Here is an update:  I have the same behavior on the server, with having the website name setted as  "site", any hint would be really great .. 

Error:
Page not found (404)
Request Method: 	POST
Request URL: 	http://mywebsite.com/newsletter/test3/subscribe/
Raised by: 	newsletter.views.SubscribeRequestView

No Newsletter matches the given query.

**Comment by frennkie** [comment]
Are you accessing `newsletter/test3/subscribe/` as an authorized users (e.g. admin)? Or is this a separate browser session?

> P.S.: another "issue" I have is copying the ‘stub’-templates to the project_dir does not change a thing, there are still the templates from the package linked

"Hiding" issue in a P.S. is not not a good idea as it might be overlooked or lead to confusion in the discussion. If this is a problem for you and you can't find a solution then please open another issue.

**Comment by frennkie** [comment]
The `127.0.0.1:8000` setting should work. 

Are you doing anything special otherwise? Is this a "clean" project or are you trying to add it to an existing site (where other apps are active that might have an impact)?

**Comment by TommyK82** [comment]
@freenkie thanks for the replay,  you are absolutly  right about the thing in the P.S.  ...   anyway this is no issue here 

I included  the form described  here: https://django-newsletter.readthedocs.io/en/latest/usage.html#embed-a-sign-up-form-within-any-page

And when I use this form for a sign up (using double opt in)  then  the  error above appears.

I just tried again with  127.0.0.1:8000   as  site  for the newsletter  test3,    no changes  still not  working,     and still working then I choose  example.com  as  site ...     

Any hints  how I can find out  the issue are  appreciated.

Yes this is  a clean project

**Comment by frennkie** [comment]
The "Site" field is a multi select list. You need to select all sites for which you want the newsletter to be active. 

If only example.com is selected, then only example.com works.

If you want others (e.g. localhost) to work the you need to select them and save.

**Comment by TommyK82** [comment]
Yes I know,  I just   tried to  say  that  the newsletter  signup  works perfectly  when I   choose  "example.com"  (really the value example.com like in the pre-settings) 

But it is not working on server (having the realswebsite as "site" or  "127.0.0.1:8000" when developing on localhost.

--> Means the newsletter  sign up is working,   the error is maybe  caused  by some missing  "wildcard" when I choose  127.0.0.1:8000  instead of  "example.com".

When email sign up is working (using  example.com as site) then it is working and I am directed  to http://127.0.0.1:8000/newsletter/smartfuse3d/subscribe/email-sent/  (which works)

When email sign up is not working (using  127.0.0.1:8000 as site)  then the error apears   http://127.0.0.1:8000/newsletter/test3/subscribe/   is not available.

So I assume  there is some setting wrong  to allow  some kind of wildcard..

**Comment by frennkie** [comment]
What do you mean by wildcard? 

Can you again post the exact settings and URL of the case that you are trying and that fails?

**Comment by TommyK82** [comment]
Really  just  a minum example check below ... many  thanks for helping ..

[mysite.zip](https://github.com/jazzband/django-newsletter/files/5684700/mysite.zip)

**Comment by woodz-** [comment]
I suggest to take #357 into a preconfigured venv. I am pretty sure the pitfall is the SITE_ID in settings.py which breaks multiple configured sites via the admin panel.
The author stated it does its job on a prod domain.

**Comment by AliMejbar** [comment]
Up I've just encountered this problem , it was working fine with sqlite but when i migrated to postgresql i started having this issue. I don't find any explanation of how it happens.
